### DISCUSSION TITLE
Is there any way to speed up the import of images into milvus

### DISCUSSION THREAD

**Discussion by zhenzi0322** [milvus-io/milvus#19211] [Q&A and General discussion]
Since I have a lot of images and corresponding image ids, is there a way to quickly convert them into vectors and add them to the milvus collection in the image similarity search service?

**Comment by yhmo** [comment]
We have a project named "Towhee", it is a python tool to convert unstructured data into embeddings:
https://github.com/towhee-io/towhee
https://towhee.io/

You can choose pre-defined models to convert embeddings, and you can even do lot of ETL works by this tool.

**Reply by zhenzi0322** [reply]
https://github.com/milvus-io/bootcamp/tree/master/solutions/reverse_image_search/quick_deploy

I use towhee, but the import within the project is done one image at a time. There is no batch import. I have a lot of images, so it's too slow to import them one by one.

![image](https://user-images.githubusercontent.com/34839719/190308636-6237c622-e937-4858-bc60-0ff601cfe8fb.png)

The current import progress is too slow.

**Reply by shiyu22** [reply]
There is some exploration on [reverse image search](https://github.com/towhee-io/examples/tree/main/image/reverse_image_search) using towhee and milvus you can refer to. For example, in the deep dive notebook, you can use `set_parallel` to accelerate.
